From 71e38b1fde105f05df93c7297cbd8ab3b621bbfc Mon Sep 17 00:00:00 2001
From: John Greene <jogreene@redhat.com>
Date: Thu, 14 Aug 2014 14:41:59 -0400
Subject: [ethernet] e1000e: Fix compilation warning when !CONFIG_PM_SLEEP

Message-id: <1408027348-9610-6-git-send-email-jogreene@redhat.com>
Patchwork-id: 87709
O-Subject: [RHEL7.1 PATCH 05/34] e1000e: Fix compilation warning when !CONFIG_PM_SLEEP
Bugzilla: 1091119
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Commit 7509963c703b (e1000e: Fix a compile flag mis-match for
suspend/resume) moved suspend and resume hooks to be available when
CONFIG_PM is set. However, it can be set even if CONFIG_PM_SLEEP is not set
causing following warnings to be emitted:

drivers/net/ethernet/intel/e1000e/netdev.c:6178:12: warning:
   'e1000_suspend' defined but not used [-Wunused-function]

drivers/net/ethernet/intel/e1000e/netdev.c:6185:12: warning:
 'e1000_resume' defined but not used [-Wunused-function]

To fix this make the hooks to be available only when CONFIG_PM_SLEEP is set
and remove CONFIG_PM wrapping from driver ops because this is already
handled by SET_SYSTEM_SLEEP_PM_OPS() and SET_RUNTIME_PM_OPS().

Signed-off-by: Mika Westerberg <mika.westerberg@linux.intel.com>
Cc: Dave Ertman <davidx.m.ertman@intel.com>
Cc: Aaron Brown <aaron.f.brown@intel.com>
Cc: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>

BZ: 1091119
Testing: Acceptable
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7834849
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/e1000e/netdev.c b/drivers/net/ethernet/intel/e1000e/netdev.c
index 295f968..58877b0 100644
--- a/drivers/net/ethernet/intel/e1000e/netdev.c
+++ b/drivers/net/ethernet/intel/e1000e/netdev.c
@@ -6196,7 +6196,7 @@ static int __e1000_resume(struct pci_dev *pdev)
  return 0;
 }
 
-#ifdef CONFIG_PM
+#ifdef CONFIG_PM_SLEEP
 static int e1000_suspend(struct device *dev)
 {
  struct pci_dev *pdev = to_pci_dev(dev);
@@ -6215,7 +6215,7 @@ static int e1000_resume(struct device *dev)
 
  return __e1000_resume(pdev);
 }
-#endif /* CONFIG_PM */
+#endif /* CONFIG_PM_SLEEP */
 
 #ifdef CONFIG_PM_RUNTIME
 static int e1000_runtime_suspend(struct device *dev)
@@ -7043,13 +7043,11 @@ static DEFINE_PCI_DEVICE_TABLE(e1000_pci_tbl) = {
 };
 MODULE_DEVICE_TABLE(pci, e1000_pci_tbl);
 
-#ifdef CONFIG_PM
 static const struct dev_pm_ops e1000_pm_ops = {
  SET_SYSTEM_SLEEP_PM_OPS(e1000_suspend, e1000_resume)
  SET_RUNTIME_PM_OPS(e1000_runtime_suspend, e1000_runtime_resume,
       e1000_idle)
 };
-#endif
 
 /* PCI Device API Driver */
 static struct pci_driver e1000_driver = {
@@ -7057,11 +7055,9 @@ static struct pci_driver e1000_driver = {
  .id_table = e1000_pci_tbl,
  .probe    = e1000_probe,
  .remove   = e1000_remove,
-#ifdef CONFIG_PM
  .driver   = {
   .pm = &e1000_pm_ops,
  },
-#endif
  .shutdown = e1000_shutdown,
  .err_handler = &e1000_err_handler
 };
-- 
1.7.1